File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-0.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int n = s.length();
        int i = 0, j = 0;
        int ans = 0;

        while (j < n) {
            boolean found = false;
            for (int k = i; k < j; k++) {
                if (s.charAt(k) == s.charAt(j)) {
                    i = k + 1;  
                    found = true;
                    break;
                }
            }

            ans = Math.max(ans, j - i + 1);
            j++;
        }

        return ans;
    }
}

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-0.cpp
class Solution {
   public:
    int characterReplacement(string s, int k) {
        int n = s.length();
        int left = 0;
        int maxi = 0;
        int ans=0;
        unordered_map<char, int> mp;
        for (int i = 0; i < n; i++) {
            mp[s[i]]++;
            maxi = max(maxi, mp[s[i]]);

            if ((i - left + 1) - maxi > k) {
                mp[s[left]]--;
                left++;
            }
            ans = max(ans, i - left + 1);
        }
        return ans;
    }
};
